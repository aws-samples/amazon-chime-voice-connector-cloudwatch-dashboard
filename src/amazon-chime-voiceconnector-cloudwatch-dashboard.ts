import { App, Duration, RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
import { Dashboard, Row } from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';
import { config } from 'dotenv';
import {
  OutboundCall,
  InboundCall,
  RoundTripTime,
  PacketLoss,
  MeanOpinionScore,
  Jitter,
  Alarms,
  LogQuery,
  SNSResources,
} from '.';

config();

const appRegions = ['us-east-1', 'us-west-2'];

export interface VoiceConnectorDashboardProps extends StackProps {
  logLevel: string;
}

export class VoiceConnectorDashboard extends Stack {
  constructor(
    scope: Construct,
    id: string,
    props: VoiceConnectorDashboardProps,
  ) {
    super(scope, id, props);

    const dashboard = new Dashboard(this, 'VoiceConnectorDashboard', {
      dashboardName: 'VoiceConnector',
      defaultInterval: Duration.minutes(60),
    });

    dashboard.applyRemovalPolicy(RemovalPolicy.DESTROY);

    const width: number = 24 / appRegions.length;
    const outboundCallRow = new Row();
    appRegions.forEach((deployRegion) => {
      new OutboundCall(this, 'outboundCallsRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: outboundCallRow,
        width: width,
      });
    });

    const inboundCallRow = new Row();
    appRegions.forEach((deployRegion) => {
      new InboundCall(this, 'inboundCallRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: inboundCallRow,
        width: width,
      });
    });

    const roundTripTimeRow = new Row();
    appRegions.forEach((deployRegion) => {
      new RoundTripTime(this, 'roundTripTimeRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: roundTripTimeRow,
        width: width,
      });
    });

    const packetLossRow = new Row();
    appRegions.forEach((deployRegion) => {
      new PacketLoss(this, 'packetLossRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: packetLossRow,
        width: width,
      });
    });

    const meanOpinionScoreRow = new Row();
    appRegions.forEach((deployRegion) => {
      new MeanOpinionScore(this, 'meanOpinionScoreRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: meanOpinionScoreRow,
        width: width,
      });
    });

    const jitterRow = new Row();
    appRegions.forEach((deployRegion) => {
      new Jitter(this, 'jitterRow_' + deployRegion, {
        metricRegion: deployRegion,
        dashboard: dashboard,
        row: jitterRow,
        width: width,
      });
    });

    // const generalLogQueryRow = new Row();
    // appRegions.forEach((deployRegion) => {
    //   new GeneralLogQuery(this, 'generalLogQueryRow_' + deployRegion, {
    //     metricRegion: deployRegion,
    //     dashboard: dashboard,
    //     row: generalLogQueryRow,
    //     width: width,
    //   });
    // });

    // const sipErrorLogQueryRow = new Row();
    // appRegions.forEach((deployRegion) => {
    //   new SipErrorLogQuery(this, 'sipErrorLogQueryRow_' + deployRegion, {
    //     metricRegion: deployRegion,
    //     dashboard: dashboard,
    //     row: sipErrorLogQueryRow,
    //     width: width,
    //   });
    // });

    dashboard.addWidgets(
      outboundCallRow,
      inboundCallRow,
      // generalLogQueryRow,
      // sipErrorLogQueryRow,
      meanOpinionScoreRow,
      roundTripTimeRow,
      packetLossRow,
      jitterRow,
    );
  }
}

export interface VoiceConnectorAlarmsProps extends StackProps {
  logLevel: string;
  deployRegion: string;
}

export class VoiceConnectorAlarms extends Stack {
  constructor(scope: Construct, id: string, props: VoiceConnectorAlarmsProps) {
    super(scope, id, props);
    const topic = new SNSResources(this, 'SNSResources-' + props.deployRegion);

    new Alarms(this, 'alarms', {
      alarmRegion: props.deployRegion,
      topic: topic.alertTopic,
    });

    new LogQuery(this, 'logQuery');
  }
}

const app = new App();

const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: 'us-east-1',
};

appRegions.forEach((deployRegion) => {
  new VoiceConnectorAlarms(app, 'VoiceConnectorAlarms-' + deployRegion, {
    logLevel: process.env.LOG_LEVEL || 'INFO',
    deployRegion: deployRegion,
    env: {
      account: process.env.CDK_DEFAULT_ACCOUNT,
      region: deployRegion,
    },
  });
});

const voiceConnectorDashboardStackProps = {
  logLevel: process.env.LOG_LEVEL || 'INFO',
};

new VoiceConnectorDashboard(app, 'VoiceConnectorDashboard', {
  ...voiceConnectorDashboardStackProps,
  env: devEnv,
});

app.synth();
