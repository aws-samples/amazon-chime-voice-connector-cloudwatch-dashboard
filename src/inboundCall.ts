import { Duration } from 'aws-cdk-lib';
import {
  Metric,
  GraphWidget,
  GraphWidgetView,
  Dashboard,
  Row,
  Stats,
} from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

interface InboundCallProps {
  metricRegion: string;
  dashboard: Dashboard;
  row: Row;
  width: number;
}

export class InboundCall extends Construct {
  public readonly widget: GraphWidget;

  constructor(scope: Construct, id: string, props: InboundCallProps) {
    super(scope, id);

    const inboundCallFailuresMetric = new Metric({
      metricName: 'InboundCallFailures',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      color: '#d62728',
      statistic: Stats.SUM,
      period: Duration.minutes(1),
    });

    const inboundCallsActiveMetric = new Metric({
      metricName: 'InboundCallsActive',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      color: '#2ca02c',
      statistic: Stats.SUM,
      period: Duration.minutes(1),
    });

    this.widget = new GraphWidget({
      left: [inboundCallsActiveMetric, inboundCallFailuresMetric],
      title: 'Inbound Calls - ' + props.metricRegion,
      liveData: true,
      width: props.width,
      view: GraphWidgetView.TIME_SERIES,
      stacked: false,
      region: props.metricRegion,
      statistic: Stats.SUM,
      period: Duration.minutes(1),
    });

    props.row.addWidget(this.widget);
  }
}
