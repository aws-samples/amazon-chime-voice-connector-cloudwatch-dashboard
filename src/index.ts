export * from './inboundCall';
export * from './jitter';
export * from './mos';
export * from './outboundCall';
export * from './packetLoss';
export * from './rtt';
export * from './alarms';
export * from './logQuery';
export * from './sns';
