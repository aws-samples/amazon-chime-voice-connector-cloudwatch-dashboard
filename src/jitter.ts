import { Duration } from 'aws-cdk-lib';
import {
  Metric,
  GraphWidget,
  MathExpression,
  GraphWidgetView,
  HorizontalAnnotation,
  Dashboard,
  Row,
  Stats,
} from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

interface JitterProps {
  metricRegion: string;
  dashboard: Dashboard;
  row: Row;
  width: number;
}

export class Jitter extends Construct {
  public readonly widget: GraphWidget;

  constructor(scope: Construct, id: string, props: JitterProps) {
    super(scope, id);

    const jitterBetweenVCandCustomerMetric = new Metric({
      metricName: 'VcToCustomerJitter',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      statistic: Stats.AVERAGE,
      period: Duration.minutes(1),
    });

    const jitterBetweenCustomerAndVCMetric = new Metric({
      metricName: 'CustomerToVcJitter',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      statistic: Stats.AVERAGE,
      period: Duration.minutes(1),
    });

    const jitterBetweenVCandCustomerMath = new MathExpression({
      expression: '(jitterBetweenVCandCustomer/100)',
      label: 'Jitter VC to Customer',
      color: '#bcbd22',
      usingMetrics: {
        jitterBetweenVCandCustomer: jitterBetweenVCandCustomerMetric,
      },
    });

    const jitterBetweenCustomerAndVCMath = new MathExpression({
      expression: '(jitterBetweenCustomerAndVC/100)',
      label: 'Jitter Customer to VC',
      color: '#e377c2',
      usingMetrics: {
        jitterBetweenCustomerAndVC: jitterBetweenCustomerAndVCMetric,
      },
    });

    const jitterHorizontalAnnotation: HorizontalAnnotation = {
      label: 'Acceptable Jitter in ms',
      value: 30,
    };
    this.widget = new GraphWidget({
      left: [jitterBetweenCustomerAndVCMath, jitterBetweenVCandCustomerMath],
      title: 'Jitter - ' + props.metricRegion,
      width: props.width,
      liveData: true,
      view: GraphWidgetView.TIME_SERIES,
      stacked: false,
      region: props.metricRegion,
      leftYAxis: {
        min: 0,
      },
      statistic: Stats.AVERAGE,
      period: Duration.minutes(1),
      leftAnnotations: [jitterHorizontalAnnotation],
    });
    props.row.addWidget(this.widget);
  }
}
