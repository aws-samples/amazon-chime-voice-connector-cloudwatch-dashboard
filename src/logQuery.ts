import { CustomResource, Duration } from 'aws-cdk-lib';
import {
  Role,
  ServicePrincipal,
  ManagedPolicy,
  PolicyDocument,
  PolicyStatement,
} from 'aws-cdk-lib/aws-iam';
import { Architecture, Runtime } from 'aws-cdk-lib/aws-lambda';
import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
import { RetentionDays } from 'aws-cdk-lib/aws-logs';
import { Provider } from 'aws-cdk-lib/custom-resources';
import { Construct } from 'constructs';

export class LogQuery extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    const logQueryCustomResourceRole = new Role(this, 'logQueryLambdaRole', {
      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
      managedPolicies: [
        ManagedPolicy.fromAwsManagedPolicyName(
          'service-role/AWSLambdaBasicExecutionRole',
        ),
      ],
      inlinePolicies: {
        ['cloudWatchPolicy']: new PolicyDocument({
          statements: [
            new PolicyStatement({
              resources: ['*'],
              actions: [
                'logs:PutQueryDefinition',
                'logs:DeleteQueryDefinition',
                'ssm:GetParameter',
                'ssm:PutParameter',
                'ssm:DeleteParameter',
                'chime:ListVoiceConnectors',
                'chime:GetVoiceConnectorLoggingConfiguration',
              ],
            }),
          ],
        }),
      },
    });

    const logQueryCustomResource = new NodejsFunction(
      this,
      'logQueryCustomResource',
      {
        handler: 'index.handler',
        entry: 'src/resources/logQuery/index.ts',
        bundling: {
          nodeModules: ['@aws-sdk/client-chime-sdk-voice'],
        },
        architecture: Architecture.ARM_64,
        timeout: Duration.minutes(1),
        runtime: Runtime.NODEJS_18_X,
        role: logQueryCustomResourceRole,
      },
    );

    const logQueryCustomResourceProvider = new Provider(
      this,
      'logQueryCustomResourceProvider',
      {
        onEventHandler: logQueryCustomResource,
        logRetention: RetentionDays.ONE_WEEK,
      },
    );

    const generalLogQuery =
      "fields @timestamp, call_id, sip_message, @logStream\n| parse sip_message '*\\n' as sip \n| display @timestamp, sip, @logStream\n| sort @timestamp desc\n| limit 50";

    new CustomResource(this, 'logQueryCustomResourceSIPMessages', {
      serviceToken: logQueryCustomResourceProvider.serviceToken,
      properties: {
        Query: generalLogQuery,
        Name: 'Voice Connector SIP Messages',
      },
    });

    const sipErrorCodesLogQuery =
      "fields @timestamp, call_id, sip_message, @logStream\n| parse sip_message '*\\n' as sip \n| filter sip like /(SIP\\/2.0 5|SIP\\/2.0 4|SIP\\/2.0 6)/\n|display @timestamp, sip, @logStream\n| sort @timestamp desc\n| limit 50";

    new CustomResource(this, 'logQueryCustomResourceSIPErrors', {
      serviceToken: logQueryCustomResourceProvider.serviceToken,
      properties: {
        Query: sipErrorCodesLogQuery,
        Name: 'Voice Connector SIP Errors',
      },
    });
  }
}
