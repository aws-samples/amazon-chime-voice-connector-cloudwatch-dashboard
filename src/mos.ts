import { Duration } from 'aws-cdk-lib';
import {
  Metric,
  GraphWidget,
  GraphWidgetView,
  HorizontalAnnotation,
  Dashboard,
  Row,
  Stats,
} from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

interface MeanOpinionScoreProps {
  metricRegion: string;
  dashboard: Dashboard;
  row: Row;
  width: number;
}
export class MeanOpinionScore extends Construct {
  public readonly widget: GraphWidget;

  constructor(scope: Construct, id: string, props: MeanOpinionScoreProps) {
    super(scope, id);

    const mosBetweenVCandCustomerMetric = new Metric({
      metricName: 'MOSBetweenVcAndCustomer',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      statistic: Stats.AVERAGE,
      period: Duration.minutes(1),
    });

    const mosHorizontalAnnotation: HorizontalAnnotation = {
      label: 'Acceptable MOS',
      value: 3.9,
    };
    this.widget = new GraphWidget({
      left: [mosBetweenVCandCustomerMetric],
      title: 'MOS - ' + props.metricRegion,
      width: props.width,
      liveData: true,
      view: GraphWidgetView.TIME_SERIES,
      stacked: false,
      region: props.metricRegion,
      leftYAxis: {
        min: 0,
        max: 5,
      },
      statistic: Stats.AVERAGE,
      period: Duration.minutes(1),
      leftAnnotations: [mosHorizontalAnnotation],
    });
    props.row.addWidget(this.widget);
  }
}
