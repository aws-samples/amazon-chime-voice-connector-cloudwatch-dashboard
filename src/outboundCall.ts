import { Duration } from 'aws-cdk-lib';
import {
  Metric,
  GraphWidget,
  GraphWidgetView,
  Dashboard,
  Row,
  Stats,
} from 'aws-cdk-lib/aws-cloudwatch';
import { Construct } from 'constructs';

interface OutboundCallProps {
  metricRegion: string;
  dashboard: Dashboard;
  row: Row;
  width: number;
}
export class OutboundCall extends Construct {
  public readonly widget: GraphWidget;

  constructor(scope: Construct, id: string, props: OutboundCallProps) {
    super(scope, id);

    const outboundCallFailuresMetric = new Metric({
      metricName: 'OutboundCallFailures',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      color: '#d62728',
      period: Duration.minutes(1),
      statistic: Stats.SUM,
    });

    const outboundCallsActiveMetric = new Metric({
      metricName: 'OutboundCallsActive',
      namespace: 'AWS/ChimeVoiceConnector',
      region: props.metricRegion,
      color: '#2ca02c',
      period: Duration.minutes(1),
      statistic: Stats.SUM,
    });

    this.widget = new GraphWidget({
      left: [outboundCallsActiveMetric, outboundCallFailuresMetric],
      title: 'Outbound Calls - ' + props.metricRegion,
      liveData: true,
      width: props.width,
      view: GraphWidgetView.TIME_SERIES,
      stacked: false,
      region: props.metricRegion,
      period: Duration.minutes(1),
      statistic: Stats.SUM,
    });
    props.row.addWidget(this.widget);
  }
}
