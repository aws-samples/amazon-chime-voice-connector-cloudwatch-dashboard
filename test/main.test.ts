import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { VoiceConnectorDashboard } from '../src/amazon-chime-voiceconnector-cloudwatch-dashboard';

const stackProps = {
  logLevel: process.env.LOG_LEVEL || 'INFO',
};

const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

test('Snapshot', () => {
  const app = new App();
  const stack = new VoiceConnectorDashboard(app, 'test', {
    ...stackProps,
    env: devEnv,
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
