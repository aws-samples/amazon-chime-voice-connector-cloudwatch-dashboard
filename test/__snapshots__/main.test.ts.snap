// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "VoiceConnectorDashboardC28983B6": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "DashboardBody": "{\\"start\\":\\"-PT1H\\",\\"widgets\\":[{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Outbound Calls - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"OutboundCallsActive\\",{\\"color\\":\\"#2ca02c\\",\\"region\\":\\"us-east-1\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"OutboundCallFailures\\",{\\"color\\":\\"#d62728\\",\\"region\\":\\"us-east-1\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Sum\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":6,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Outbound Calls - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"OutboundCallsActive\\",{\\"color\\":\\"#2ca02c\\",\\"region\\":\\"us-west-2\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"OutboundCallFailures\\",{\\"color\\":\\"#d62728\\",\\"region\\":\\"us-west-2\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Sum\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":12,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Inbound Calls - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"InboundCallsActive\\",{\\"color\\":\\"#2ca02c\\",\\"region\\":\\"us-east-1\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"InboundCallFailures\\",{\\"color\\":\\"#d62728\\",\\"region\\":\\"us-east-1\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Sum\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":18,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Inbound Calls - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"InboundCallsActive\\",{\\"color\\":\\"#2ca02c\\",\\"region\\":\\"us-west-2\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"InboundCallFailures\\",{\\"color\\":\\"#d62728\\",\\"region\\":\\"us-west-2\\",\\"period\\":60,\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Sum\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":24,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"MOS - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"MOSBetweenVcAndCustomer\\",{\\"region\\":\\"us-east-1\\",\\"period\\":60}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable MOS\\",\\"value\\":3.9,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"max\\":5}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":30,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"MOS - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[\\"AWS/ChimeVoiceConnector\\",\\"MOSBetweenVcAndCustomer\\",{\\"region\\":\\"us-west-2\\",\\"period\\":60}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable MOS\\",\\"value\\":3.9,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"max\\":5}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":36,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"RTT - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"RTT in ms\\",\\"expression\\":\\"(rttBetweenVCandCustomer/1000)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"RTTBetweenVcAndCustomer\\",{\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"rttBetweenVCandCustomer\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable RTT in ms\\",\\"value\\":150,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":42,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"RTT - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"RTT in ms\\",\\"expression\\":\\"(rttBetweenVCandCustomer/1000)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"RTTBetweenVcAndCustomer\\",{\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"rttBetweenVCandCustomer\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable RTT in ms\\",\\"value\\":150,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":48,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Packet Loss % - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"Packet Lost % - CustomerToVC\\",\\"color\\":\\"#e377c2\\",\\"expression\\":\\"IF(vcToCustomerRtpPackets, IF(vcToCustomerPacketsLost, (3000-vcToCustomerRtpPackets)/3000*100, 0), 0)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcRtpPackets\\",{\\"label\\":\\"CustomerToVcRtpPackets\\",\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"customerToVcRtpPackets\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcPacketsLost\\",{\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"customerToVcPacketsLost\\"}],[{\\"label\\":\\"Packet Lost % - VCToCustomer\\",\\"color\\":\\"#bcbd22\\",\\"expression\\":\\"IF(vcToCustomerRtpPackets, IF(vcToCustomerPacketsLost, (3000-vcToCustomerRtpPackets)/3000*100, 0), 0)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerRtpPackets\\",{\\"label\\":\\"VcToCustomerRtpPackets\\",\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"vcToCustomerRtpPackets\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerPacketsLost\\",{\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"vcToCustomerPacketsLost\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"AcceptablePacket Loss\\",\\"value\\":1,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":54,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Packet Loss % - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"Packet Lost % - CustomerToVC\\",\\"color\\":\\"#e377c2\\",\\"expression\\":\\"IF(vcToCustomerRtpPackets, IF(vcToCustomerPacketsLost, (3000-vcToCustomerRtpPackets)/3000*100, 0), 0)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcRtpPackets\\",{\\"label\\":\\"CustomerToVcRtpPackets\\",\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"customerToVcRtpPackets\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcPacketsLost\\",{\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"customerToVcPacketsLost\\"}],[{\\"label\\":\\"Packet Lost % - VCToCustomer\\",\\"color\\":\\"#bcbd22\\",\\"expression\\":\\"IF(vcToCustomerRtpPackets, IF(vcToCustomerPacketsLost, (3000-vcToCustomerRtpPackets)/3000*100, 0), 0)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerRtpPackets\\",{\\"label\\":\\"VcToCustomerRtpPackets\\",\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"vcToCustomerRtpPackets\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerPacketsLost\\",{\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"vcToCustomerPacketsLost\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"AcceptablePacket Loss\\",\\"value\\":1,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":0,\\"y\\":60,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Jitter - us-east-1\\",\\"region\\":\\"us-east-1\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"Jitter Customer to VC\\",\\"color\\":\\"#e377c2\\",\\"expression\\":\\"(jitterBetweenCustomerAndVC/100)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcJitter\\",{\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"jitterBetweenCustomerAndVC\\"}],[{\\"label\\":\\"Jitter VC to Customer\\",\\"color\\":\\"#bcbd22\\",\\"expression\\":\\"(jitterBetweenVCandCustomer/100)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerJitter\\",{\\"region\\":\\"us-east-1\\",\\"visible\\":false,\\"id\\":\\"jitterBetweenVCandCustomer\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable Jitter in ms\\",\\"value\\":30,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}},{\\"type\\":\\"metric\\",\\"width\\":12,\\"height\\":6,\\"x\\":12,\\"y\\":66,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Jitter - us-west-2\\",\\"region\\":\\"us-west-2\\",\\"stacked\\":false,\\"metrics\\":[[{\\"label\\":\\"Jitter Customer to VC\\",\\"color\\":\\"#e377c2\\",\\"expression\\":\\"(jitterBetweenCustomerAndVC/100)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"CustomerToVcJitter\\",{\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"jitterBetweenCustomerAndVC\\"}],[{\\"label\\":\\"Jitter VC to Customer\\",\\"color\\":\\"#bcbd22\\",\\"expression\\":\\"(jitterBetweenVCandCustomer/100)\\"}],[\\"AWS/ChimeVoiceConnector\\",\\"VcToCustomerJitter\\",{\\"region\\":\\"us-west-2\\",\\"visible\\":false,\\"id\\":\\"jitterBetweenVCandCustomer\\"}]],\\"annotations\\":{\\"horizontal\\":[{\\"label\\":\\"Acceptable Jitter in ms\\",\\"value\\":30,\\"yAxis\\":\\"left\\"}]},\\"yAxis\\":{\\"left\\":{\\"min\\":0}},\\"liveData\\":true,\\"period\\":60,\\"stat\\":\\"Average\\"}}]}",
        "DashboardName": "VoiceConnector",
      },
      "Type": "AWS::CloudWatch::Dashboard",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
